src/common/snippets/src/port_descriptor.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/port_descriptor.hpp"
+#include "ngraph/except.hpp"
+#include <sstream>
+
+namespace ngraph {
+namespace snippets {
+
+size_t PortDescriptor::Scheduling::FULL_DIM = SIZE_MAX;
+
+PortDescriptor::PortDescriptor(const ov::Input<ov::Node>& in, std::vector<size_t> subtensor_shape, std::vector<size_t> layout)
+        : PortDescriptor(ov::Input<const Node>(in.get_node(), in.get_index()), std::move(subtensor_shape), std::move(layout)) {}
+PortDescriptor::PortDescriptor(const ov::Input<const ov::Node>& in, std::vector<size_t> subtensor_shape, std::vector<size_t> layout)
+        : PortDescriptor(in.get_shape(), std::move(subtensor_shape), std::move(layout)) {}
+
+PortDescriptor::PortDescriptor(const ov::Output<ov::Node>& out, std::vector<size_t> subtensor_shape, std::vector<size_t> layout)
+        : PortDescriptor(ov::Output<const Node>(out.get_node(), out.get_index()), std::move(subtensor_shape), std::move(layout)) {}
+PortDescriptor::PortDescriptor(const ov::Output<const ov::Node>& out, std::vector<size_t> subtensor_shape, std::vector<size_t> layout)
+        : PortDescriptor(out.get_shape(), std::move(subtensor_shape), std::move(layout)) {}
+
+PortDescriptor::PortDescriptor(std::vector<size_t> shape, std::vector<size_t> subtensor_shape, std::vector<size_t> layout)
+    : m_tensor_shape(std::move(shape)), m_layout(std::move(layout)), m_subtensor_shape(std::move(subtensor_shape)) {
+    validate_arguments();
+}
+
+void PortDescriptor::validate_arguments() {
+    if (!m_tensor_shape.empty() && m_layout.empty()) {
+        m_layout.resize(m_tensor_shape.size());
+        // NCHW layout by default
+        std::iota(m_layout.begin(), m_layout.end(), 0);
+    } else if (m_layout.size() != m_tensor_shape.size()) {
+        OPENVINO_THROW("Snippets tensor descriptor: Layout size must be equal to the shape size");
+    }
+}
+
+PortDescriptor PortDescriptor::deserialize(const std::string& serialized_info) {
```

Comment:
Minor for discussion: do we really need deserialization of Port descriptor? If we do, than is it sufficient to enable correct serialization-deserialization of the subgraph?

Reply:
I don't know because I just moved it from `TensorDescriptor` too. I don't know for what it, to be honest. I removed it.
Moreover, I don't understand why we need serialization :)